### ISSUE TITLE
allow students to share work

### ISSUE THREAD

**Issue by phaustin** [eoas-ubc/docker_by_example#2]
tracking: https://github.com/jupyterhub/jupyterhub/issues/394

https://github.com/pangeo-data/pangeo/issues/680

https://blog.danallan.com/posts/2019-07-01-sharing-jupyter-notebooks/

**Comment by slemonide** [comment]
Just a static notebook? Can just upload it to gitlab/github. It would then even be possible to read notebooks in the browser without downloading them.

Alternatively, we can have shared volumes between students.

If it's just code, without large data files, git solution seems cleaner, as then it would be easy for students to actually collaborate, and not just share.

**Comment by phaustin** [comment]
yes, but then we have to teach them git, which is probably overkill at the beginning.  Shared volumes would be better for paired programming when they were starting out in breakout rooms, then we could introduce them to git for projects later in the term.

**Comment by slemonide** [comment]
I am not sure if jupyter would work nicely with shared volumes. It doesn't have any support for real-time collaborative editing.

CoCalc seems to have support for that though.

**Comment by slemonide** [comment]
By shared volumes I meant having it writable only by one student, and readable by everyone. That would avoid possible problems with different jupyter instances trying to write to the same file at the same time.

**Comment by phaustin** [comment]
yes -- looks like subclassing dockerspawner is the official way to do it:

https://github.com/jupyterhub/dockerspawner/issues/172
